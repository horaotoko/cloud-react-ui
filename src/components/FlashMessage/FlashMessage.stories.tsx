import { text, radios } from "@storybook/addon-knobs";
import { FlashMessage as FlashMessageComponent, Props } from "./FlashMessage";

export default {
  component: FlashMessageComponent,
  title: "components/Flash Message",
};

const colorOptions = {
  success: "success",
  warning: "warning",
  error: "error",
};

export const FlashMessage = (): JSX.Element => (
  <FlashMessageComponent
    color={radios("Color", colorOptions, "success") as Props["color"]}
  >
    {text("Text", "Flash Message")}
  </FlashMessageComponent>
);
