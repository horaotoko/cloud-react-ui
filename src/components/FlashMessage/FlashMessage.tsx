import { forwardRef } from "react";
import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";

export type Props = {
  color?: "success" | "warning" | "error";
  className?: string;
  children: React.ReactNode;
};

const StyledFlashMessage = styled.div<Props>`
  ${({ color = "success", theme: { flashMessage } }) => css`
    width: ${flashMessage.width};
    padding: ${flashMessage.padding};
    background-color: ${flashMessage.backgroundColor[color]};
    border-width: ${flashMessage.borderWidth};
    border-style: ${flashMessage.borderStyle};
    border-color: ${flashMessage.borderColor[color]};
    border-radius: ${flashMessage.borderRadius};
    color: ${flashMessage.color[color]};
    font-size: ${flashMessage.fontSize};
  `}
`;
StyledFlashMessage.defaultProps = defaultProps;

const FlashMessage = forwardRef<HTMLDivElement, Props>((props, ref) => (
  <StyledFlashMessage ref={ref} {...props} />
));

FlashMessage.displayName = "FlashMessage";

export { FlashMessage };
