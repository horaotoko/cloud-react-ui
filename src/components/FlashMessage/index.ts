export { FlashMessage } from "./FlashMessage";
export type { Props as FlashMessageProps } from "./FlashMessage";
