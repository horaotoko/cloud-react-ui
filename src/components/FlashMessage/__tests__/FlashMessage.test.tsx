import { render } from "@testing-library/react";
import { FlashMessage } from "../FlashMessage";

describe("FlashMessage", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <FlashMessage color="success">flash message</FlashMessage>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
