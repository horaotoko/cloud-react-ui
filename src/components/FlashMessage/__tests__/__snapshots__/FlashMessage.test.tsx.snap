// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessage snapshot test 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  padding: 12px;
  background-color: #E6EAED;
  border-width: 1px;
  border-style: solid;
  border-color: #A9BCDB;
  border-radius: 4px;
  color: #4772B3;
  font-size: 12px;
}

<div
    class="c0"
    color="success"
  >
    flash message
  </div>
</DocumentFragment>
`;
